cli.js

```diff
@@ -28,7 +34,20 @@ program
       program.args = []
       return
     }
-    const options = {}
+    const options = {
+      parser: (buffer) => {
```

Comment:
I'm not sure about this myself yet, buy my gut reaction is I wish this function was pulled out of the option parsing logic so what was left here read `parser: decompressIfCompressed`. That looks like a really long name but maybe you can think of a more elegant one.